File: PersionalExpenses/Domain/Category.cs
namespace PersionalExpenses.Domain;

public enum CategoryType { Income = 0, Expense = 1 }

public class Category
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public string Name { get; set; } = default!;
    public CategoryType Type { get; set; }
    public long? ParentId { get; set; }
    public string? Color { get; set; }
    public string? Icon { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsDefault { get; set; } = false;

    public User User { get; set; } = default!;
    public Category? Parent { get; set; }
    public ICollection<Category> Children { get; set; } = new List<Category>();
    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: PersionalExpenses FE/src/types/category.ts
export interface Category {
  id: number;
  name: string;
  type: number; // 0 income, 1 expense (assumption)
  parentId: number | null;
  color: string; // hex color like #00e5e9
  icon: string; // emoji or icon string
  isActive: boolean;
  isDefault: boolean;
}

export interface CreateCategoryRequest {
  name: string;
  type: number;
  parentId?: number | null;
  color?: string;
  icon?: string;
}

export interface UpdateCategoryRequest extends Partial<CreateCategoryRequest> {}
